.github/workflows/unit-tests.yml

```diff
@@ -0,0 +1,34 @@
+name: Ruby-2.3
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+jobs:
+  test:
+
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        ruby: [2.3, 2.4, 2.5]
+
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set up Ruby
+      uses: ruby/setup-ruby@ec106b438a1ff6ff109590de34ddc62c540232e0
+      with:
+        ruby-version: ${{ matrix.ruby }}
+      env:
+        ACTIONS_ALLOW_UNSECURE_COMMANDS: true
+    - name: Add hammer-cli-foreman to local gem file
+      run: echo "gem 'hammer_cli_foreman', :git => 'https://github.com/theforeman/hammer-cli-foreman.git'" > Gemfile.local
+    - name: Add hammer-cli to local gem file
+      run: echo "gem 'hammer_cli', :git => 'https://github.com/theforeman/hammer-cli.git'" >> Gemfile.local
+    - name: Install dependencies
+      run: bundle install
+    - name: Bundle update
+      run: gem update bundler
+    - name: Run tests
+      run: COVERALLS_REPO_TOKEN=${{ secrets.COVERALLS_REPO_TOKEN }} bundle exec rake
```

Comment:
Updated to use `pull_request_target`

Reply:
@jeremylenz with `pull_request_target` the checks don't run anymore 😞 I am fine with just having it push the events to coveralls on a `push` if you are?

Reply:
I think we need to have unit tests run.  If coveralls fails it's not ideal but we can live with it.  But I think we will need to switch back to `pull_request`.

Can we split this into two workflows-- one for `pull_request` that runs the unit tests and ignores Coveralls, and one for `push` that just runs the coveralls?

Reply:
@jeremylenz 

The tests are kicked off from here:

https://github.com/Katello/hammer-cli-katello/blob/master/Rakefile#L28

Which uses this file then:

https://github.com/Katello/hammer-cli-katello/blob/master/test/test_helper.rb

So we can't just do the coveralls unless I am not reading it correctly, or are you saying just have the push do this line:
`COVERALLS_REPO_TOKEN=${{ secrets.COVERALLS_REPO_TOKEN }} bundle exec rake`

and the other one just do `bundle exec rake`

Reply:
Will `bundle exec rake` without the token run the unit tests and pass the check?  (on a `pull_request` event) If so then that looks like a good solution to me.  I just want to be sure unit tests pass or fail the check properly for PR's.

Reply:
Yeah it will run the tests, and if they fail, exit out with the test failure results like we have in Katello. 

It will try to send the event but without the token it will just say repo not found, so nothing will actually get transmitted. 

I can change it back to `pull_request` and make one of the tests fail to show it's working.